File: python/wiringpi_smp01.py
import wiringpi
from wiringpi import GPIO
import time

wiringpi.wiringPiSetup()
# wiringpi.pinMode(2, GPIO.OUTPUT)
# wiringpi.digitalWrite(2, GPIO.LOW)
wiringpi.pinMode(2, GPIO.INPUT)
wiringpi.pullUpDnControl(2, GPIO.PUD_DOWN)
while(True):
    pin_value = wiringpi.digitalRead(2)
    print('pin_value = ' + str(pin_value))
    time.sleep(1)

File: c_cpp/wiringpi_smp01/wiringpi_smp01.c
#include <wiringPi.h>
#include <stdio.h>

int main (void)
{
    unsigned char pin_value;
    wiringPiSetup();
//    pinMode(2, OUTPUT);
//    digitalWrite(2, LOW);
    pinMode(2, INPUT);
    pullUpDnControl(2, PUD_UP);
    while(TRUE) {
        pin_value = digitalRead(2);
        printf("pin_value = %d\n",pin_value);

        delay(1000);
    }

    return 0;
}
